### DISCUSSION TITLE
Defining State of charge

### DISCUSSION THREAD

**Discussion by wranson22** [pybamm-team/PyBaMM#2889] [Q&A]
Hello, I'm looking for some assistance with defining the state of charge (SOC) after a simulation runs. 
In my mind, SOC is a percentage of the battery that is able to discharged. for example: If I discharge 50 Ah out of a 100 Ah battery, I'd want the SOC to return a value around 50.  
In my mind, this would be found with a Remaining_capacity[A.h]/Battery_capacity[A.h]
In addition, I haven't figured out how to consider the fact that it's a Lead-acid battery, that typically shouldn't be discharged below 50%. the Lower cutoff voltage seems like it's in place to keep the battery from killing itself.  
Would it be reasonable to take the range of the cutoff voltages and assign 10% intervals linearly across the voltage spread? 

The application of my code is to run for 1 hour, charge or discharge the batteries, and then return the SOC to be used as the initial state of charge the next time it runs. 

I've looked at the discharge capacity variable in the output, but I'm not entirely sure how to interpret it. Any assistance would be appreciated. 

[battery_code_042023.txt](https://github.com/pybamm-team/PyBaMM/files/11290877/battery_code_042023.txt)

**Comment by wranson22** [comment]
Anyone able to point me in the right direction on this?

**Comment by brosaplanella** [comment]
Different users might define SoC differently, that's why we don't have a standardised variable for it. You can define your own version of SoC from the variable `"Discharge capacity [A.h]"`. [This notebook](https://colab.research.google.com/github/pybamm-team/PyBaMM/blob/main/examples/notebooks/Getting%20Started/Tutorial%206%20-%20Managing%20simulation%20outputs.ipynb) shows how to manipulate the outputs of a simulation.

**Reply by wranson22** [reply]
That makes sense. I think I just don't fully understand how the discharge capacity variable behaves. It goes positive when the battery is discharging, and negative when it's charging. Would you have any insight into how I can define the SoC such that it is a scalar value that goes between 0 and 1 based on if the battery is fully charged (1) or fully discharged(0)
I would greatly appreciate any guidance in that process. Project due very soon and need to get that part working

**Reply by valentinsulzer** [reply]
That's because initial conditions aren't necessarily at 100% SOC, so it's possible to discharge and then charge back above 100%. You could just apply a linear mapping to discharge capacity so that SOC goes between 0 and 1 for any particular test. 

Specifically for lead-acid, `solution["X-averaged electrolyte concentration [mol.m-3]"]/c_max`, where `c_max` is some maximum value you can choose, is also a decent proxy for SOC. For example, choose `c_max` to be the maximum observed value of x-averaged electrolyte concentration, so that SOC maxes out at 1

**Reply by wranson22** [reply]
I see, so then the Parameter `'Initial State of Charge'` is not a scalar that operates between 0 and 1? I saw the value assigned in the parameter set was 1.0 so I made that assumption. My original idea was to run this code multiple times, after each iteration, return the `SoC` value, and on the next iteration, use that `SoC` as the `'Initial State of Charge'`
 
Is this something that you think will work? Or is there a new approach I need to take?
